### PR TITLE
Fix for empty example and empty default value

### LINKED ISSUE #3882: Swagger UI always displays empty default and example keys
**API Platform version(s) affected**: 2.6.0 alpha, maybe 2.78

**Description**  

image

**How to reproduce**  

Use this entity: https://github.com/api-platform/api-platform/blob/master/api/src/Entity/Greeting.php

**Possible Solution**  

Check for empty values here: https://github.com/api-platform/core/blob/master/src/JsonSchema/SchemaFactory.php#L195-L201

### PR DESCRIPTION
**Pull request by Tayfun74**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tickets       | fixes #3882
| License       | MIT

### CODE REVIEW

File: src/JsonSchema/SchemaFactory.php [hunk 1]
```diff
@@ -192,11 +192,11 @@ private function buildPropertySchema(Schema $schema, string $definitionName, str
             $propertySchema['externalDocs'] = ['url' => $iri];
         }
 
-        if (!isset($propertySchema['default']) && null !== $default = $propertyMetadata->getDefault()) {
```
Comment (dunglas) [inline src/JsonSchema/SchemaFactory.php:195]: You can remove the test for `null` values, it's already handled by `empty`.

**Review by dunglas — CHANGES_REQUESTED** [review]
Could you also add a test please?

**Review by dunglas — APPROVED** [review]

**Comment by dunglas** [comment]
Thank you @Tayfun74!

**Comment by drupol** [comment]
@Tayfun74 I always try to never use `empty()` in my code because it's confusing.

I'm definitely sharing this point of view here: https://twitter.com/localheinz/status/1335638838279069696

**Pull request merged**